debezium-core/src/test/java/io/debezium/transforms/ExtractNewRecordStateTest.java

```diff
@@ -336,6 +336,36 @@ public void testAddFields() {
         }
     }
 
+    @Test
+    @FixFor({ "DBZ-2606" })
+    public void testNewFieldAndHeaderMapping() {
+        try (final ExtractNewRecordState<SourceRecord> transform = new ExtractNewRecordState<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(ADD_FIELDS, "op:OP, lsn:LSN, id:ID, source.lsn, transaction.total_order:TOTAL_ORDER");
+            props.put(ADD_HEADERS, "op:OP,source.lsn:SOURCE_LSN,transaction.id:TXN_ID,transaction.total_order:TOTAL_ORDER");
+            props.put(ADD_HEADERS_PREFIX, "prefix.");
+            transform.configure(props);
+
+            final SourceRecord updateRecord = createUpdateRecord();
+            final SourceRecord unwrapped = transform.apply(updateRecord);
+            assertThat(((Struct) unwrapped.value()).get("__OP")).isEqualTo(Envelope.Operation.UPDATE.code());
```

Comment:
The Kafka connect `Struct` is not aware of debezium field prefixes. I think we will have to hard-code here.

Reply:
Sorry, I'm not following. It's a convention by this Debezium SMT to prefix any additional field with "__". I.e. we can control to add or not add that prefix in case of _renamed_ additional fields. My suggestion is to remove this, i.e. if a user wants to rename `id` to `__ID`, they'd have to put `id:__ID` into the config. Does that make sense?

Reply:
Ok. I understand.
Then I need to also document that `add.headers.prefix` and `add.fields.prefix` only when given will be taken into consideration for this new field name mapping. Is my understanding correct? I will fix that

Reply:
Ok, so I think I retract what I said, because I forgot that this prefix actually is configurable already via those options you mention. I thought the `__` would be hard-coded, and there'd be no way to get rid of it, but there actually is: you can set it to ``; I think so, at least. In that light, it seems fair to apply any given prefix (either the default one, or any custom one) also in case the property is renamed. I.e. what you got here is reasonable. Sorry for the forth and back, needed to recap all the details at first.

Reply:
Thanks. So, i will not do any changes. 
When the mapping is given and the `add.headers.prefix`  and `add.fields.prefix` is not set, then it will be defaulted to __ (double underscore).
When the user want to set a different prefix (other than the default __ (double underscore)), he will explicitly do so.

Can you please accept this pull request? Thanks a lot.